### PR TITLE
Added Profile resource

### PR DESCRIPTION
**Pull request by lydakis**
Description of changes:
Added AWS::Transfer::Profile resource, implemented handlers, and performed contract tests.

### Testing
- mvn verify
- sam tests
- contract tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-transfer-profile/aws-transfer-profile.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    "CertificateIds": {
+      "description": "An array of certificate id to apply to this resource.",
+      "type": "array",
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/CertificateId"
+      }
+    },
+    "Arn": {
+      "description": "Specifies the unique Amazon Resource Name (ARN) for the agreement.",
```
Comment (mayurm88) [inline aws-transfer-profile/aws-transfer-profile.json:70]: `Specifies the unique Amazon Resource Name (ARN) for the profile.`

Comment (vicwuqinyi) [reply aws-transfer-profile/aws-transfer-profile.json:70]: ++

Comment (lydakis) [reply aws-transfer-profile/aws-transfer-profile.json]: Good catch, updated!

File: aws-transfer-profile/pom.xml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>transfer</artifactId>
+            <version>2.17.259</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/software.amazon.awssdk/sdk-core -->
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>sdk-core</artifactId>
+            <version>2.17.136</version>
+        </dependency>
```
Comment (mayurm88) [inline aws-transfer-profile/pom.xml:34]: While testing this resource type by registering it privately in my AWS account, I constantly got an error while trying to provision the profile. Following is the error I encountered:

```
Lambda function handler threw an uncaught exception: IS_NONE_AUTH_TYPE_REQUEST
```

Tried multiple things and ultimately, changing this dependency to be of same version as the transfer client worked.
```suggestion
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>sdk-core</artifactId>
            <version>2.17.259</version>
        </dependency>
```

Comment (lydakis) [reply aws-transfer-profile/pom.xml:34]: Ahh thank you, good catch!

**Review by kwonminj — APPROVED** [review]

**Review by j-schoeff — APPROVED** [review]
Internally reviewed. LGTM.

**Pull request merged**